application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -4603,6 +4603,54 @@ input TransferPreferencesInput {
   slack: Duration
 }
 
+"""
+A collection of selectors for what routes/agencies should be included in or excluded from the search.
+
+The `include` is always applied to select the initial set, then `exclude` to remove elements.
+If only `exclude` is present, the exclude is applied to the existing set of results.
+
+Therefore, if an entity is both included _and_ excluded the exclusion takes precedence.
+"""
+input TransitFilterInput {
+  """
+  A list of selectors for what routes/agencies should be allowed during search.
+    
+  In order to be excluded a route/agency has to match with at least one selector.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  exclude: [TransitFilterSelectInput!]
+  """
+  A list of selectors for what routes/agencies should be excluded during the search.
+    
+  If route/agency matches with at least one selector it will be included.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  include: [TransitFilterSelectInput!]
+}
+
+"""
+A list of selectors for including or excluding entities from the routing results. Null
+means everything is allowed to be returned and empty lists are not allowed.
+
+An entity will match with selectors if it matches with all lists.
```

Comment:
What lists do you refer to here? On this level, it's not possible to define multiple lists anymore (for now) due to the `@oneOf` directive.